cmd/cosign/cli/triangulate.go

```diff
@@ -52,17 +53,28 @@ func MungeCmd(ctx context.Context, imageRef string) error {
 		return err
 	}
 
-	// TODO: just return the descriptor directly if we have a digest reference.
-	desc, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain), remote.WithContext(ctx))
-	if err != nil {
-		return err
+	// If the image ref contains the digest, use it.
+	// Otherwise, look up the digest the tag currently points to.
+	var h v1.Hash
```

Comment:
I don't like that we have this same thing 4 places :( could we factor it out somehow?

Reply:
For you Dan, anything.

Reply:
What about for golangci-lint though?
![image](https://user-images.githubusercontent.com/1714486/127056119-1313a7a2-fc03-4e3c-830f-4dfbbe090cbe.png)